src/pathfinding.ts

```diff
@@ -0,0 +1,173 @@
+        let neighbor: Coordinates;
+        // 1st possible neighbor
+        neighbor = { y: y, x: x + 1 };
+        createEdgePairForNeighbor(
+          gameState,
+          hexBoardGraph,
+          stone_color,
+          current_cell,
+          neighbor
+        );
+        // 2nd possible neighbor
+        neighbor = { y: y + 1, x: x };
+        createEdgePairForNeighbor(
+          gameState,
+          hexBoardGraph,
+          stone_color,
+          current_cell,
+          neighbor
+        );
+        // 3rd possible neighbor
+        neighbor = { y: y + 1, x: x - 1 };
+        createEdgePairForNeighbor(
+          gameState,
+          hexBoardGraph,
+          stone_color,
+          current_cell,
+          neighbor
+        );
+      }
+    });
+  });
+
+  // Add edges for start and end nodes
+  gameState.board.forEach((_, i) => {
+    addEdge(hexBoardGraph, "black-start", `${i}-0`);
+    addEdge(hexBoardGraph, `${i}-${gameState.board.length - 1}`, "black-end");
+    addEdge(hexBoardGraph, "white-start", `0-${i}`);
+    addEdge(hexBoardGraph, `${gameState.board.length - 1}-${i}`, "white-end");
+  });
+}
+
+function createEdgePairForNeighbor(
+  gameState: GameState,
+  hexBoardGraph: HexBoardGraph,
+  stone_color: "black",
+  current_cell: Coordinates,
+  neighbor: Coordinates
+) {
+  if (cellExistsAndHasStone(gameState, neighbor, stone_color)) {
+    addEdgePair(
+      hexBoardGraph,
+      `${current_cell.y}-${current_cell.x}`,
+      `${neighbor.y}-${neighbor.x}`
+    );
+  }
+}
+
+function cellExistsAndHasStone(
+  gameState: GameState,
+  cell: Coordinates,
+  stone_color: "black"
+): boolean {
+  if (cell.y < 0 || cell.y >= gameState.board.length) {
+    return false;
+  }
+  if (cell.x < 0 || cell.x >= gameState.board[cell.y].length) {
+    return false;
+  }
+  return gameState.board[cell.y][cell.x].value == stone_color;
+}
+
+function addVertex(hexBoardGraph: HexBoardGraph, id: string) {
+  hexBoardGraph.vertexMap.set(id, hexBoardGraph.graph.addVertex(id));
+}
+
+function addEdge(
+  hexBoardGraph: HexBoardGraph,
+  vertex1: string,
+  vertex2: string
+) {
+  hexBoardGraph.graph.addEdge(
+    hexBoardGraph.vertexMap.get(vertex1),
+    hexBoardGraph.vertexMap.get(vertex2),
+    1
+  );
+}
+
+function addEdgePair(
+  hexBoardGraph: HexBoardGraph,
+  vertex1: string,
+  vertex2: string
+) {
+  hexBoardGraph.graph.addEdgePair(
+    hexBoardGraph.vertexMap.get(vertex1),
+    hexBoardGraph.vertexMap.get(vertex2),
+    1
+  );
+}
+
+function existsPathForPlayer(hexBoardGraph: HexBoardGraph): boolean {
+  const dijkstra = new jkstra.algos.Dijkstra(hexBoardGraph.graph);
+
+  const path_black = dijkstra.shortestPath(
+    hexBoardGraph.vertexMap.get("black-start"),
+    hexBoardGraph.vertexMap.get("black-end")
+  );
+  const path_white = dijkstra.shortestPath(
+    hexBoardGraph.vertexMap.get("white-start"),
+    hexBoardGraph.vertexMap.get("white-end")
+  );
+
+  return !!(path_black || path_white);
```

Comment:
Yes, the `!!` operator ensures I get a boolean as the function output. I had to use this trick, because the jkstra lib returns `null` if no path is found, and apparenlty `null || null` equals `null` in JS, which is not what I want